test/entrypoint/srv_with_hotreload.lua

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env tarantool
+
+require('strict').on()
+
+local log = require('log')
+local errors = require('errors')
+local cartridge = require('cartridge')
+errors.set_deprecation_handler(function(err)
+    log.error('%s', err)
+    os.exit(1)
+end)
+
+local ok, err = errors.pcall('CartridgeCfgError', cartridge.cfg, {
+    roles = {
+        'cartridge.roles.metrics',
+    },
+    roles_reload_allowed = true,
```

Comment:
Why not simply add it to existing srv_basic?

Reply:
To preserve compatibility with cartridge < 2.3.0 use

```lua
roles_reload_allowed = os.getenv('TARANTOOL_ROLES_RELOAD_ALLOWED'),
```

And pass it as an env var from the test